pkg/plugins/plugins.go

```diff
@@ -187,8 +188,9 @@ func Handle(iface string, fn func(string, *Client)) {
 	extpointHandlers[iface] = fn
 }
 
-// GetAll returns all the plugins for the specified implementation
-func GetAll(imp string) ([]*Plugin, error) {
+// GetAll returns all the plugins for the specified implementation that have
+// not already been registered
+func GetAll(imp string, ext map[string]volume.Driver) ([]*Plugin, error) {
```

Comment:
Instead of passing in a list to filter, why not compare to the list we already have in this package?